andela/noldor-ah-backend
# #161290909 Add Tagging Functionality

#### What does this PR do?
This PR adds tagging functionality to the Authors Haven API. Tagging can be done when creating a new article or when editing an existing article.

- Add tagging functionality
- Retrieve tags when getting a single article
- Add tests to ensure that the feature works as expected

NOTE: 
1. Tags should be separated with commas
2. Tags cannot consist of only numbers
3. Tags cannot contain special characters (except '-')

#### Description of Task to be completed?
- `/PUT 200 OK - /articles/:slug/tags`
    --- Required field: `tags`

#### How should this be manually tested?
1. Clone the repo and run npm i
2. Create a `.env` file using the `.env.sample` example provided
3. Create a new user #11 and copy the `x-token` returned
4. Add a field called tag when creating a new article #13 using the copied `x-token` as authorization
5. Retrieve the recently created article and you'll see a `tags` key in the response object containing the tags you specified when creating the article

##### Tagging an existing article
1. Make a `PUT` request to `/articles/:slug/tags` replacing `:slug` with the slug of the article you want to update
2. Add a field called `tags` containing the tags you want to add to the article in your request body
3. You should get a success message when done correctly

#### Any background context you want to provide?
Tags that do not already exist in the database will be created when tagging an article.

#### What are the relevant pivotal tracker stories?
#161290909

#### Screenshots (if appropriate)
<img width="1123" alt="screen shot 2018-11-07 at 1 35 49 pm" src="https://user-images.githubusercontent.com/36507972/48131946-18901b00-e292-11e8-9814-32d52a4b0848.png">
<img width="1124" alt="screen shot 2018-11-07 at 1 31 46 pm" src="https://user-images.githubusercontent.com/36507972/48131886-e54d8c00-e291-11e8-85a0-de94da150502.png">
<img width="1127" alt="screen shot 2018-11-07 at 1 33 04 pm" src="https://user-images.githubusercontent.com/36507972/48131897-eda5c700-e291-11e8-85d8-157b325f6238.png">


#### Questions:
N/A
